src/main/java/edu/harvard/iq/dataverse/DatasetPage.java

```diff
@@ -4283,4 +4296,29 @@ public String getJsonLd() {
         }
         return "";
     }
+
+    public void selectAllFiles() {
+        logger.info("selectAllFiles called");
+        List<FileMetadata> allFiles = new ArrayList<>();
+        for (FileMetadata fm : workingVersion.getFileMetadatas()) {
+            allFiles.add(fm);
+        }
+        selectedFiles = allFiles;
+    }
+
+    public void clearSelection() {
+        logger.info("clearSelection called");
+        selectedFiles = Collections.EMPTY_LIST;
+    }
+    
+    public void fileListingPaginatorListener(PageEvent event) {
+        setFilePaginatorPage(event.getPage());      
+    }
+    
+    public void refreshPaginator() {
+        FacesContext facesContext = FacesContext.getCurrentInstance();
+        org.primefaces.component.datatable.DataTable dt = (org.primefaces.component.datatable.DataTable) facesContext.getViewRoot().findComponent("datasetForm:tabView:filesTable");
```

Comment:
importing `DataTable` would make this line much clearer. ie. `DataTable dt = (DataTable) facesContext.getViewRoot().findComponent("datasetForm:tabView:filesTable");`

Reply:
The problem here is that there's already a DataTable object defined in dataverse so the ugliness is in line is related to making the distinction with the native DataTable object. If there's a cleaner way to make the distinction...

Reply:
Oh yeah, that sucks. You could import up to `component`. It's not ideal, but I think better.

Reply:
how I wish for python's `import x as y`

Reply:
This isn't worth changing.